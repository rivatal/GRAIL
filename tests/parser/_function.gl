func(a, b) {
    c = 0;
    return a + b + c;
}

func(1, 2);
func(a, b);