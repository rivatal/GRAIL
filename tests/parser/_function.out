Func(int func (Formal(int, a), Formal(int, b)) {Var_dec(Local(int, c, Num_Lit(0)));
Return(Binop(Binop(Id(a), Add, Id(b)), Add, Id(c)));})
Expr(Call(func, [Num_Lit(1), Num_Lit(2)]));
Expr(Call(func, [Id(a), Id(b)]));